File: 2025/1.lua
local B = arg[1] == "b"
local p = 50
local zeroes = 0

for rot, dist in io.read("a"):gmatch("([LR])(%d+)") do
    dist = tonumber(dist)
    for i = 1, dist do
        p = (p + (rot == "L" and -1 or 1)) % 100
        if p == 0 and (((i == dist) and not B) or B) then
            zeroes = zeroes + 1
        end
    end
end

print(zeroes)

File: 2025/1.py
import sys

B = len(sys.argv) > 1 and sys.argv[1] == "b"
p = 50
zeroes = 0

for line in sys.stdin.readlines():
    rot, dist = line[0], int(line[1:])
    for i in range(dist):
        p += (-1 if rot == "L" else 1)
        p %= 100
        if p == 0 and ((i == dist-1 and not B) or B):
            zeroes += 1

print(zeroes)